fiftyone/brain/internal/core/leaky_splits.py

```diff
@@ -115,312 +147,241 @@ def method(self):
+                    leak_ids.append(unique_id)
                     break
 
-        return self.samples.select([sample_id] + neighbors_ids)
+        return self.samples.select([sample_id] + leak_ids, ordered=True)
+
+    def no_leaks_view(self, view=None):
+        """Returns a view with leaks excluded.
+
+        Args:
+            view (None): an optional :class:`fiftyone.core.view.DatasetView`
+                from which to exclude. By default, :meth:`samples` is used
+        """
+        if self._thresh is None:
+            raise ValueError("You must first call `find_leaks()`")
+
+        if view is None:
+            view = self.samples
 
-    def no_leaks_view(self, view):
-        return view.exclude(self.leaks_view().values("id"))
+        return view.exclude(self._leak_ids)
 
     def tag_leaks(self, tag="leak"):
-        """Tag leaks"""
-        self.leaks_view().tag_samples([tag])
+        """Tags all potential leaks in :meth:`leaks_view` with the given tag.
 
-    def _id2splitConstructor(self):
+        Args:
+            tag ("leak"): the tag string to apply
+        """
+        self.leaks_view().tag_samples(tag)
 
-        # do this once at the beggining of the run
-        # has O(n) memory cost but memory is cheap compared to
-        # doing a couple of `in` operations per sample every run
-        # I want that sweet sweet O(1) lookup
+    def _initialize(self):
         id2split = {}
+
+        split_ids = {}
         for split_name, split_view in self.split_views.items():
-            sample_ids = split_view.values("id")
+            sample_ids = set(split_view.values("id"))
+            split_ids[split_name] = sample_ids
             id2split.update({sid: split_name for sid in sample_ids})
 
-        return id2split
-
-
-def _to_views(samples, split_views=None, split_field=None, split_tags=None):
-    """Helper function so that we can always work with views. I.e. creates
-    the splits denoted by one of the keyword arguments as views.
-
-    Args:
-        - samples: the samples in the datset
-        One and only one of the following:
-        - split_views (None): a dict of the splits as views. If this value is passed
-        it is returned as is
-        - split_field (None): a string corresponding to a field that has a unique
-        string value for each split e.g. samples that are in the 'train' split
-        will have a value of 'train' in this split.
-        - split_tags (None): a list of strings corresponding to the tags of the
-        different splits.
-    """
-
-    arithmetic_true = lambda x: int(x is not None)
-    num_given = (
-        arithmetic_true(split_views)
-        + arithmetic_true(split_field)
-        + arithmetic_true(split_tags)
-    )
+        # Check for overlapping splits
+        split_names = list(split_ids.keys())
+        for idx, split1 in enumerate(split_names):
+            for split2 in split_names[idx + 1 :]:
+                overlap = split_ids[split1] & split_ids[split2]
```

Comment:
You might have had a check like this in the initial implementation but I cautioned against it because loading all IDs into memory via `values("id")` can potentially be expensive for large datasets.

I added it back for two reasons:
- The implementation *already* requires calling `values("id")` for all splits, so might as well do the validation
- I wasn't aware that the implementation requires `DuplicatesMixin`, which is only implemented by the sklearn backend, which isn't really designed for large datasets anyway